src/OrleansAzureUtils/MultiClusterNetwork/AzureTableBasedGossipChannel.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans.Runtime.Configuration;
+using Orleans.MultiCluster;
+
+namespace Orleans.Runtime.MultiClusterNetwork
+{
+    // An implementation of a gossip channel based on a  standard orleans azure table
+    // multiple gossipnetworks can use the same table, and are separated by pkey = GlobalServiceId
+    internal class AzureTableBasedGossipChannel : IGossipChannel
+    {
+
+        private TraceLogger logger;
+        private GossipTableInstanceManager tableManager;
+        private static int seqno;
+
+        public string Name { get; private set; }
+
+        public async Task Initialize(GlobalConfiguration globalconfig, string connectionstring)
+        {
+            Name = "AzureTableBasedGossipChannel-" + ++seqno;
+            logger = TraceLogger.GetLogger(Name, TraceLogger.LoggerType.Runtime);
+
+            logger.Info("Initializing Gossip Channel for ServiceId={0} using connection: {1}, SeverityLevel={2}",
+                globalconfig.GlobalServiceId, ConfigUtilities.RedactConnectionStringInfo(connectionstring), logger.SeverityLevel);
+
+            tableManager =
+                await GossipTableInstanceManager.GetManager(globalconfig.GlobalServiceId, connectionstring, logger);
+        }
+
+        // used by unit tests
+        public Task DeleteAllEntries()
+        {
+            logger.Info("DeleteAllEntries");
+            return tableManager.DeleteTableEntries();
+        }
+
+        public struct Pair<L,R>
```

Comment:
why the need for this type and not just using Tuple? If it's because Tuple is a reference type and you actually need a value type, then how about KeyValuePair?